api/query/atoms.go

```diff
@@ -704,70 +799,110 @@ func MetadataQualityFromString(quality string) (MetadataQuality, error) {
 }
 
 func unmarshalQ(b []byte) (AbstractQuery, error) {
-	var tnp TestNamePattern
-	err := json.Unmarshal(b, &tnp)
-	if err == nil {
-		return tnp, nil
-	}
-	var stnp SubtestNamePattern
-	err = json.Unmarshal(b, &stnp)
-	if err == nil {
-		return stnp, nil
-	}
-	var tp TestPath
-	err = json.Unmarshal(b, &tp)
-	if err == nil {
-		return tp, nil
-	}
-	var tse TestStatusEq
-	err = json.Unmarshal(b, &tse)
-	if err == nil {
-		return tse, nil
-	}
-	var tsn TestStatusNeq
-	err = json.Unmarshal(b, &tsn)
-	if err == nil {
-		return tsn, nil
-	}
-	var n AbstractNot
-	err = json.Unmarshal(b, &n)
-	if err == nil {
-		return n, nil
-	}
-	var o AbstractOr
-	err = json.Unmarshal(b, &o)
-	if err == nil {
-		return o, nil
-	}
-	var a AbstractAnd
-	err = json.Unmarshal(b, &a)
-	if err == nil {
-		return a, nil
-	}
-	var e AbstractExists
-	err = json.Unmarshal(b, &e)
-	if err == nil {
-		return e, nil
-	}
-	var s AbstractSequential
-	err = json.Unmarshal(b, &s)
-	if err == nil {
-		return s, nil
-	}
-	var c AbstractCount
-	err = json.Unmarshal(b, &c)
-	if err == nil {
-		return c, nil
-	}
-	var l AbstractLink
-	err = json.Unmarshal(b, &l)
-	if err == nil {
-		return l, nil
-	}
-	var i MetadataQuality
-	err = json.Unmarshal(b, &i)
-	if err == nil {
-		return i, nil
+	{
+		var tnp TestNamePattern
+		err := json.Unmarshal(b, &tnp)
+		if err == nil {
+			return tnp, nil
+		}
+	}
+	{
+		var stnp SubtestNamePattern
+		err := json.Unmarshal(b, &stnp)
+		if err == nil {
+			return stnp, nil
+		}
+	}
+	{
+		var tp TestPath
+		err := json.Unmarshal(b, &tp)
+		if err == nil {
+			return tp, nil
+		}
+	}
+	{
+		var tse TestStatusEq
+		err := json.Unmarshal(b, &tse)
+		if err == nil {
+			return tse, nil
+		}
+	}
+	{
+		var tsn TestStatusNeq
+		err := json.Unmarshal(b, &tsn)
+		if err == nil {
+			return tsn, nil
+		}
+	}
+	{
+		var n AbstractNot
+		err := json.Unmarshal(b, &n)
+		if err == nil {
+			return n, nil
+		}
+	}
+	{
+		var o AbstractOr
+		err := json.Unmarshal(b, &o)
+		if err == nil {
+			return o, nil
+		}
+	}
+	{
+		var a AbstractAnd
+		err := json.Unmarshal(b, &a)
+		if err == nil {
+			return a, nil
+		}
+	}
+	{
+		var e AbstractExists
+		err := json.Unmarshal(b, &e)
+		if err == nil {
+			return e, nil
+		}
+	}
+	{
+		var a AbstractAll
```

Comment:
All / None re-used a/n, and the variables persisting makes for copy-pasta bugs, so I've scoped all the attempts. This (AbstractAll) and AbstractNone are added.